theano/gpuarray/neighbours.py

```diff
@@ -13,7 +13,7 @@
 
 from .basic_ops import (as_gpuarray_variable, GpuKernelBase, Kernel,
                         infer_context_name)
-from .opt import register_opt as register_gpu_opt, op_lifter
+from .opt import register_opt2, op_lifter, register_opt
```

Comment:
This is bad. You have renabmed one import. I think it should be 

from .opt import register_opt2, op_lifter, register_opt as register_gpu_opt